python/pyspark/sql/pandas/serializers.py

```diff
@@ -1223,8 +1223,17 @@ def dump_stream(self, iterator, stream):
         Read through an iterator of (iterator of pandas DataFrame), serialize them to Arrow
         RecordBatches, and write batches to stream.
         """
-        result = [(b, t) for x in iterator for y, t in x for b in y]
-        super().dump_stream(result, stream)
+
```

Comment:
nit: linter asked me to have an empty line here - probably due to inner function?